### PR TITLE
docs: test examples

### PR DESCRIPTION
**Pull request by byashimov**
Adds the ability to use examples for tests.

- examples are tested
- easier to maintain yaml code in yaml files rather than in go
- adds multiple examples support
- dry: reuses examples to run tests
- uses service integration/endpoint examples for tests, other kinds remain in go code

Recommended to review with [whitespaces off](https://github.com/aiven/aiven-operator/pull/723/files?diff=unified&w=1).

### CODE REVIEW

**Comment by byashimov** [comment]
How it looks like:
image

**Comment by byashimov** [comment]
[tests](https://github.com/aiven/aiven-operator/actions/runs/8935831614)

**Comment by byashimov** [comment]
Blocked by https://github.com/aiven/aiven-operator/pull/724

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ markdown_extensions:
         - docs/.snippets
   - pymdownx.tilde
   - pymdownx.details
+  - pymdownx.magiclink
```
Comment (rriski) [inline docs/mkdocs.yml:22]: Where do we use this plugin?

**Review by rriski — APPROVED** [review]

File: docs/mkdocs.yml (hunk 1 above)
Comment (byashimov) [reply docs/mkdocs.yml:22]: > Where do we use this plugin?

It turns plain text urls into a tag 🙂

**Pull request merged**